internal/pkg/util/fs/fuse/fuse_mount_linux.go

```diff
@@ -142,6 +147,7 @@ func (i *ImageMount) generateCmdArgs() ([]string, error) {
 	if i.AllowOther {
 		opts = append(opts, "allow_other")
 	}
+	opts = append(opts, i.ExtraMountOpts...)
```

Comment:
We should check that `ExtraMountOpts` does not contain any options that are handled by other fields.

E.g. `ro/rw` `nosetuid` etc. should not be permitted as an extra option. We need to be able to rely on the fact that e.g. inspecting `.ReadOnly` or `.AllowSetuid` gives us the true state of the mount.

This is potentially critical for security and should be tested.